akka-actor/src/main/scala/akka/event/Logging.scala

```diff
@@ -1228,6 +1268,385 @@ trait DiagnosticLoggingAdapter extends LoggingAdapter {
   def clearMDC(): Unit = mdc(emptyMDC)
 }
 
+final class LogMarker(val name: String)
+object LogMarker {
+  /** The Marker is internally transferred via MDC using using this key */
+  private[akka] final val MDCKey = "marker"
+
+  def apply(name: String): LogMarker = new LogMarker(name)
+  /** Java API */
+  def create(name: String): LogMarker = apply(name)
+
+  def extractFromMDC(mdc: MDC): Option[String] =
+    mdc.get(MDCKey) match {
+      case Some(v) ⇒ Some(v.toString)
+      case None    ⇒ None
+    }
+
+}
+
+/**
+ * [[LoggingAdapter]] extension which adds Marker support.
+ * Only recommended to be used within Actors as it isn't thread safe.
+ */
+trait MarkerLoggingAdapter extends DiagnosticLoggingAdapter {
+
+  /**
+   * Log message at error level, including the exception that caused the error.
+   * @see [[LoggingAdapter]]
+   */
+  def error(marker: LogMarker, cause: Throwable, message: String): Unit =
+    if (isErrorEnabled) {
+      val pre = mdc
+      mdc(pre.updated(LogMarker.MDCKey, marker.name))
```

Comment:
I don't like this dance with using `mdc` to pass the marker, since you anyway created the x3 classes.
MDC has a major disadvantage in that it's not thread safe.
Can we let `MarkerLoggingAdapter extends LoggingAdapter` instead, so that `MarkerLoggingAdapter` is also thread safe?